scripts/utils/bdlClassnameManager

```diff
@@ -0,0 +1,225 @@
+
+    Please note that this tool will not catch all cases, so reviewing changes may be required.
+*/
+
+const conversionMap = {
+    // 'btn': 'bdl-Button',
+    // 'btn-group': 'bdl-ButtonGroup',
+    // 'btn-plain': 'bdl-Button--plain',
+    // 'btn-primary': 'bdl-Button--primary',
+    // 'is-disabled': 'bdl-is-disabled',
+    // 'label': 'bdl-Label',
+    // 'pill': 'bdl-Pill',
+    // 'pill-error': 'bdl-Pill--error',
+    // 'select-button': 'bdl-SelectButton',
+    // 'toggle': 'bdl-Toggle',
+    // 'tooltip': 'bdl-Tooltip',
+};
+const scssPrefix = '.';
+const jsPrefixes = ['\'', '"', '`'];
+
+function lengthSorter (a, b) {
+    return a.length > b.length ? -1 : 1;
+}
+
+function printHelp() {
+    console.log(`Usage: ${process.argv[1].split('/').pop()} operation fileName [--verbose]`);
+    console.log('\nSupported operations:');
+    console.log('\tcheck\t\tcheck to see if a file contains deprecated values (returns error if it does)');
+    console.log('\tswap\t\treplace known-deprecated variable names in a JS or SCSS file');
+}
+
+function printVerboseOperationMessage(fileName, matches) {
+    console.log(colors.cyan('File name:'), colors.white(fileName));
+    console.log(colors.cyan('Found:'), matches);
+}
+
+async function handleWriteFile(matches, fileName, fileToParse, verboseMode) {
+    try {
+        if (matches.length > 0) {
+            await writeFile(fileName, fileToParse);
+            console.log(colors.cyan(`Replaced ${matches.length} classname value${matches.length !== 1 ? 's': ''} in:\t`), fileName);
+        } else if (verboseMode) {
+            console.warn(colors.yellow('No changes needed to file:\t'), fileName);
+        }
+    } catch (error) {
+        console.error(colors.red('Cannot write updated file to disk\n'), error);
+        process.exit(1);
+    }
+}
+
+async function main() {
+    // argv 0 and 1 are the node instance and the script name respectively.
+    // argv 2 is the operation: check, swap, append, extend
+    const operation = process.argv[2];
+
+    // argv 3 is the file (and extension)
+    const fileName = process.argv[3];
+
+    if (!fileName) {
+        console.error(colors.red('Missing parameter:'), colors.white('fileName'));
+        process.exit(1);
+    }
+    if (!Object.keys(conversionMap).length) {
+        console.error(colors.red('Missing values in: '), colors.white('conversionMap'));
+        process.exit(1);
+    }
+    const fileType = fileName.split('.').pop();
+    const verboseMode = process.argv[4] === '--verbose';
+    const extendMode = operation === 'extend'
+    const currentConversionMap = extendMode ? invert(conversionMap) : conversionMap;
+
+    let reBadName;
+    if (fileType === 'scss') {
+        const converter = entry => `\\${scssPrefix}${entry}(?!-)`;
+        reBadName = new RegExp(
+            Object.keys(currentConversionMap)
+                .map(converter)
+                .sort(lengthSorter) // sort order matters here for regex
```

Comment:
Comments are super helpful at providing context. Why does sort matter here?

Reply:
I'll make the comment more clear -- the sort order means that the regex will find matches for longer classnames first, before root classname matches. I couldn't find another way to have multiple options for exact matches in a single regex.

For example:
```
// bad
'.btn .btn-primary'.match(/\.btn|\.btn-primary/g)
=> [".btn", ".btn"]

// good
'.btn .btn-primary'.match(/\.btn-primary|\.btn/g)
=> [".btn", ".btn-primary"]
```